### PR TITLE
feat(infrastructure): add web socket service

### PR DESCRIPTION
**Pull request by tassiluca**

### CODE REVIEW

File: infrastructure/src/main/scala/io/github/positionpal/location/infrastructure/ws/WebSockets.scala
Comment (giovaz94) [inline infrastructure/src/main/scala/io/github/positionpal/location/infrastructure/ws/WebSockets.scala:1]: Nice Implementation!
One thing that I've implemented in my service, and maybe you want  also to integrate in your, are the versioned routes. The implementation is simply straightforward, its based on the concept of RoutesProvider

```
/** Trait representing a RoutesProvider: an entity that contains the versioned endpoints for a webserver */
trait RoutesProvider:
  /** A [[String]] representing the version of the routes */
  def version: String

  /** Return the routes for the following provider */
  def routes: Route

  /** Return the versioned routes for the following provider */
  def versionedRoutes: Route =
    pathPrefix(version):
      routes
```

And an example of its implementation is the following:

```
private case class ApplicationV1Routes(system: ActorSystem[?]) extends RoutesProvider:
    override def routes: Route = webSocketFlowRoute
    override def version: String = "v1"

    /** Routes used for handling the websocket
      * @return The route where the clients connect to the server and exchanges messages using websocket
      */
    private def webSocketFlowRoute: Route =
      // val actorName = s"websocket-${java.util.UUID.randomUUID().toString}"
      // val incomingActorReference = system.systemActorOf(incomingHandler, actorName)
      pathPrefix("messages" / Segment): groupId =>
        parameter("user"): userId =>
          handleWebSocketMessages(WebSocketHandlers.testingWsHandler)

  /** Return the routes for the v1 api version
    * @param system the implicit actor system
    * @return A [[Route]] object containing the routes of the server.
    */
  def v1Routes(using system: ActorSystem[?]): Route = ApplicationV1Routes(system).versionedRoutes
```
Then the user will be able to navigate to the routes with links conformed like this one: ws://127.0.0.1:8080/v1/messages/123?user=134

Comment (giovaz94) [reply infrastructure/src/main/scala/io/github/positionpal/location/infrastructure/ws/WebSockets.scala:1]: Note that's not a mandatory feature.

File: presentation/src/main/scala/io/github/positionpal/location/presentation/ModelCodecs.scala
Comment (giovaz94) [inline presentation/src/main/scala/io/github/positionpal/location/presentation/ModelCodecs.scala:1]: why changing the implementation from ArrayBased to MapBased codecs?

**Review by giovaz94 — APPROVED** [review]
Green light for me

File: presentation/src/main/scala/io/github/positionpal/location/presentation/ModelCodecs.scala
Comment (tassiluca) [reply presentation/src/main/scala/io/github/positionpal/location/presentation/ModelCodecs.scala:1]: I use borer also for json se/deserialization and MapBased codecs allow me to create well-formed json (with key - value pairs)

File: infrastructure/src/main/scala/io/github/positionpal/location/infrastructure/ws/WebSockets.scala
Comment (tassiluca) [reply infrastructure/src/main/scala/io/github/positionpal/location/infrastructure/ws/WebSockets.scala:1]: That's nice! I'll include this also for location service! Thanks 😁

**Pull request merged**